plugins/item_previews/web_client/views/ItemPreviewWidget.js

```diff
@@ -1,103 +1,179 @@
-import ItemCollection from 'girder/collections/ItemCollection';
+import _ from 'underscore';
+import $ from 'jquery';
 import View from 'girder/views/View';
-import { restRequest } from 'girder/rest';
+import { restRequest, apiRoot } from 'girder/rest';
+import { _whenAll } from 'girder/misc';
 
 import ItemPreviewWidgetTemplate from '../templates/itemPreviewWidget.pug';
+import ItemPreviewItemTemplate from '../templates/itemPreviewItem.pug';
 import '../stylesheets/itemPreviewWidget.styl';
 
 /**
  * The Item Preview widget shows a preview of items under a given folder.
- * For now, the only supported item previews are image previews.
+ * For now, this only supports json and images file types.
  */
 var ItemPreviewWidget = View.extend({
     events: {
         'click a.g-item-preview-link': function (event) {
             var id = this.$(event.currentTarget).data('id');
             this.parentView.itemListView.trigger('g:itemClicked', this.collection.get(id));
-        },
-        'click .g-widget-item-previews-wrap': function (event) {
-            this.wrapPreviews = !this.wrapPreviews;
-            this.render();
-            if (this.wrapPreviews) {
-                this.$('.g-widget-item-previews-container').addClass('g-wrap');
-            } else {
-                this.$('.g-widget-item-previews-container').removeClass('g-wrap');
-            }
         }
     },
 
     initialize: function (settings) {
-        this.collection = new ItemCollection();
+        this._isSupportedItem = this._isSupportedItem.bind(this);
+        this.debouncedaddMoreItems = _.debounce(this.addMoreItems, 100);
+        this.onScroll = _.debounce(this.onScroll, 100);
 
-        this.collection.on('g:changed', function () {
-            this.trigger('g:changed');
-            this.render();
-        }, this).fetch({
-            folderId: settings.folderId
-        });
+        this.initialized = false;
+
+        this.addingMoreItems = false;
 
-        this.wrapPreviews = false;
+        this.supportedItems = [];
+    },
+
+    // Because the parent-child life cycle is not clean, this method is needed.
+    setCollection: function (collection) {
+        this.collection = collection;
+        this.collection.on('g:changed', () => {
+            this.supportedItems = this.collection.toJSON().filter(this._isSupportedItem);
```

Comment:
Backbone Collections natively have a `.filter(` method. Using `.toJSON` will cause the contained Models to be converted to ordinary objects, losing their special methods.

See  http://backbonejs.org/#Collection-Underscore-Methods